kv/bulk: TestKvBuf failed

kv/bulk.TestKvBuf [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4633991&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4633991&tab=artifacts#/) on master @ [54671e7befc13ef0d4885c8506831bb0c6182ffc](https://github.com/cockroachdb/cockroach/commits/54671e7befc13ef0d4885c8506831bb0c6182ffc):


```
=== RUN   TestKvBuf
    kv_buf_test.go:68: 
        	Error Trace:	kv_buf_test.go:68
        	Error:      	Not equal: 
        	            	expected: false
        	            	actual  : true
        	Test:       	TestKvBuf
--- FAIL: TestKvBuf (0.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKvBuf.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-14023

kv/bulk.TestKvBuf [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4633995&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4633995&tab=artifacts#/) on master @ [54671e7befc13ef0d4885c8506831bb0c6182ffc](https://github.com/cockroachdb/cockroach/commits/54671e7befc13ef0d4885c8506831bb0c6182ffc):


```
=== RUN   TestKvBuf
    kv_buf_test.go:68: 
        	Error Trace:	kv_buf_test.go:68
        	Error:      	Not equal: 
        	            	expected: false
        	            	actual  : true
        	Test:       	TestKvBuf
--- FAIL: TestKvBuf (2.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKvBuf.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/bulk/kv_buf_test.go
+++ b/pkg/kv/bulk/kv_buf_test.go
@@ -63,7 +63,7 @@ func TestKvBuf(t *testing.T) {
 	b := kvBuf{}
 	for i := range src {
 		size := sz(len(src[i].key) + len(src[i].value))
-		fits := len(b.entries) < cap(b.entries) && len(b.slab)+int(size) < cap(b.slab)
+		fits := len(b.entries) <= cap(b.entries) && len(b.slab)+int(size) <= cap(b.slab)
 
 		require.Equal(t, fits, b.fits(ctx, size, 0, &none))
 		if !fits {

